polars/polars-lazy/src/dsl/mod.rs

```diff
@@ -301,22 +302,7 @@ impl Expr {
 
     /// Drop NaN values
     pub fn drop_nans(self) -> Self {
-        self.apply(
-            |s| match s.dtype() {
-                DataType::Float32 => {
-                    let ca = s.f32()?;
-                    let mask = ca.is_not_nan();
-                    ca.filter(&mask).map(|ca| ca.into_series())
-                }
-                DataType::Float64 => {
-                    let ca = s.f64()?;
-                    let mask = ca.is_not_nan();
-                    ca.filter(&mask).map(|ca| ca.into_series())
-                }
-                _ => Ok(s),
-            },
-            GetOutput::same_type(),
-        )
+        self.map_private(NanFunction::DropNans.into(), "drop_nans")
```

Comment:
Note that `apply` has different aggregation behavior than `map`. As the number of rows are influenced by this operation we must use `apply` here otherwise we get flawed results/oob access.

Reply:
updated to use `apply_private` instead of `map_private`